src/main/java/com/amazonaws/codedeploy/AWSCodeDeployPublisher.java

```diff
@@ -232,14 +232,14 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
     }
 
     private FilePath getSourceDirectory(FilePath basePath) throws IOException, InterruptedException {
-        String subdirectory = this.subdirectory.trim().length() > 0 ? this.subdirectory.trim() : "";
+        String subdirectory = StringUtils.trimToEmpty(this.subdirectory);
         if (!subdirectory.isEmpty() && !subdirectory.startsWith("/")) {
             subdirectory = "/" + subdirectory;
         }
         FilePath sourcePath = basePath.withSuffix(subdirectory).absolutize();
-        File sourceDirectory = new File(sourcePath.getRemote());
-        if (!sourceDirectory.isDirectory() || !isSubDirectory(basePath, sourcePath)) {
-            throw new IllegalArgumentException("Provided path is not a subdirectory of the workspace: " + sourcePath );
+        if (!sourcePath.isDirectory() || !isSubDirectory(basePath, sourcePath)) {
```

Comment:
Yes, it could be false if subdirectory contains something like "../non-workspace-dir". I guess that's what the original author wanted to check.